libcxx/utils/ci/buildkite-pipeline.yml

```diff
@@ -28,7 +28,7 @@ env:
     # LLVM RELEASE bump remove compiler ToT - 3, e.g. "Clang 15"
     LLVM_STABLE_VERSION: "16"
     LLVM_HEAD_VERSION: "17"
-    GCC_STABLE_VERSION: "12"
+    GCC_STABLE_VERSION: "13"
```

Comment:
@philnik777 do we need to backport this. I think this is not really required.

The same regarding the param.py changes. These changes also contain parts not of the original patch.

Reply:
I think we want to test the compiler which we claim to support, so IMO we should backport this. What exactly is in this patch that was not in the original?

Reply:
> I think we want to test the compiler which we claim to support, so IMO we should backport this. What exactly is in this patch that was not in the original?

Still I prefer to keep backporting as small as possible. I feel changing the compiler is more  a feature than a bug fix.

You enabled more warnings. For example, -Wno-unknown-pragmas"

Reply:
If we don’t cherry pick this, I think it means we are broken on GCC in the release branch. If so, I think we should either cherry pick this or a slimmer version of it, but we should fix the is_constructible thing in the diff at least.

Last word goes to @mordante though since he’s managing this release for libc++.

Reply:
Agreed. My comment is related to the backporting of the CI changes not the patch in general.

When we branched the release we supported GCC-12 so I think we should keep that working in the release branch. It's not our policy to keep "GCC-release - 1" working, but we did when we branched. So removing and not testing that in the release branch may break people that were happy with the previous LLVM-17 rc versions.

Reply:
Oh, that’s what I missed. Yes in that case we should have a custom patch for release/17.x thats just fixes gcc 13 compilation. I thought we didn’t support gcc 12 before the branch.